plugins/modules/system/java_keystore.py

```diff
@@ -199,23 +205,33 @@ def create_jks(module, name, openssl_bin, keytool_bin, keystore_path, password):
             if os.path.exists(keystore_p12_path):
                 os.remove(keystore_p12_path)
 
-            export_p12_cmd = "%s pkcs12 -export -name '%s' -in '%s' -inkey '%s' -out '%s' -passout 'pass:%s'" % (
-                openssl_bin, name, certificate_path, private_key_path, keystore_p12_path, password)
-            (rc, export_p12_out, export_p12_err) = run_commands(module, export_p12_cmd)
+            export_p12_cmd = [openssl_bin, "pkcs12", "-export", "-name", name, "-in", certificate_path,
+                              "-inkey", private_key_path, "-out",
+                              keystore_p12_path, "-passout", "stdin"]
+
+            # when keypass is provided, add -passin
+            cmd_stdin = ""
+            if keypass:
+                export_p12_cmd.append("-passin")
+                export_p12_cmd.append("stdin")
+                cmd_stdin = "%s\n" % keypass
+
+            cmd_stdin += "%s\n%s" % (password, password)
```

Comment:
Don't you need a final return?
```suggestion
            cmd_stdin += "%s\n%s\n" % (password, password)
```
Maybe it is not needed (probably EOF is making OpenSSL happy), but I think having it is nicer.

Reply:
Ansible is adding it in `lib/ansible/module_utils/basic.py` if `data` is passed to `run_command` and it is not binary. Adding it here, would cause a double `\n` to be passed via stdin.